packages/perseus/src/mixins/widget-prop-denylist.ts

```diff
@@ -23,6 +23,7 @@ const denylist = [
     "apiOptions",
     "questionCompleted",
     "findWidgets",
+    "hintMode",
```

Comment:
Stuff like this makes me nervous that adding new props to widgets is not an entirely trivial process :/

Reply:
Question: How did you know this needed to get added here? Is that what made you nervous? That there are steps involved that we may not be fully aware of or that these steps even exist? I'm guessing the latter as I write this out, but could be both :)

Reply:
I searched the codebase for one of the other props in this list, with the goal of finding places where widget props are created. That led me to this file.

Tangent:

Looking more closely at how this list is used, I see: https://github.com/Khan/perseus/blob/7bf32b73395a8bc14f9198ce183de0e7706bee86/packages/perseus/src/widgets/passage-ref.tsx#L79-L83

This function is supposed to return a `UserInput`, so perhaps what we ought to be doing here is just creating an object that has the fields in `UserInput`. Then we could remove the denylist.

```ts
getUserInput: () => UserInput = () => ({
  static: this.props.static,
  reviewModeRubric: this.props.reviewModeRubric,
  linterContext: this.props.linterContext,
  isLastUsedWidget: this.props.isLastUsedWidget,
  alignment: this.props.alignment,
});
```

However, a quick search makes it seem like the return value of `getUserInput` is not actually used (it's passed to `PassageRef.validate()`, but that function doesn't do anything with it and just returns a hardcoded value). So I think maybe we can just delete all this code.

Reply:
Okay, `getUserInput` _is_ used, by the Renderer (which calls it on each widget).

Reply:
😂 Got excited about potentially just deleting it all. It looks like the denylist is used in other places as well, but `removeDenylistProps` only seems to be used with the `getUserInput` here. Is the `getUserInput` used by Renderer this same method from PassageRef? In any case, probably still good to add `hintMode` here if it should be excluded when widgets serialize themselves. Thanks for looking into this!

Reply:
`getUserInput` is used as part of the scoring system. Each widget provides a `simpleValidate()` function that the Renderer calls. This function scores the widget in its current state. 

`simpleValidate` typically calls the `getUserInput` instance function on itself. This function returns the input(s) that the user has provided (the selected radio option(s), the graph coordinate that the user has moved the point to, etc).  The widget then calls a `validate()` static method on itself and passes the user input and the "rubric" (the widget options that define the correct answer) and returns the score (correct/incorrect). 

The `removeDenylistProps` in this flow is a cheat and I'd love to get rid of it. Basically, it returns all widget props as the user input, except those that show up in the `denylist`.  

Instead `getUserInput` on each widget should just return an object that represents the minimum necessary set of values (from props) that are needed by `simpleValidate`.